ts/src/woofipro.ts

```diff
@@ -0,0 +1,2636 @@
+        const orderQtyKey = isStop ? 'quantity' : 'order_quantity';
+        const priceKey = isStop ? 'price' : 'order_price';
+        if (price !== undefined) {
+            request[priceKey] = this.priceToPrecision (symbol, price);
+        }
+        if (amount !== undefined) {
+            request[orderQtyKey] = this.amountToPrecision (symbol, amount);
+        }
+        params = this.omit (params, [ 'stopPrice', 'triggerPrice', 'takeProfitPrice', 'stopLossPrice', 'trailingTriggerPrice', 'trailingAmount', 'trailingPercent' ]);
+        let response = undefined;
+        if (isStop) {
+            response = await this.v1PrivatePutAlgoOrder (this.extend (request, params));
+        } else {
+            const clientOrderId = this.safeStringN (params, [ 'clOrdID', 'clientOrderId', 'client_order_id' ]);
+            params = this.omit (params, [ 'clOrdID', 'clientOrderId', 'client_order_id' ]);
+            if (clientOrderId !== undefined) {
+                request['client_order_id'] = clientOrderId;
+            }
+            // request['side'] = side.toUpperCase ();
+            // request['symbol'] = market['id'];
+            response = await this.v1PrivatePutOrder (this.extend (request, params));
+        }
+        //
+        // {
+        //     "success": true,
+        //     "timestamp": 1702989203989,
+        //     "data": {
+        //       "status": "EDIT_SENT"
+        //     }
+        // }
+        //
+        const data = this.safeDict (response, 'data', {});
+        data['timestamp'] = this.safeInteger (response, 'timestamp');
+        return this.parseOrder (data, market);
+    }
+
+    async cancelOrder (id: string, symbol: Str = undefined, params = {}) {
+        /**
+         * @method
+         * @name woofipro#cancelOrder
+         * @see https://orderly.network/docs/build-on-evm/evm-api/restful-api/private/cancel-order
+         * @see https://orderly.network/docs/build-on-evm/evm-api/restful-api/private/cancel-order-by-client_order_id
+         * @see https://orderly.network/docs/build-on-evm/evm-api/restful-api/private/cancel-algo-order
+         * @see https://orderly.network/docs/build-on-evm/evm-api/restful-api/private/cancel-algo-order-by-client_order_id
+         * @description cancels an open order
+         * @param {string} id order id
+         * @param {string} symbol unified symbol of the market the order was made in
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {boolean} [params.stop] whether the order is a stop/algo order
+         * @param {string} [params.clientOrderId] a unique id for the order
+         * @returns {object} An [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        const stop = this.safeBool2 (params, 'stop', 'trigger', false);
+        params = this.omit (params, [ 'stop', 'trigger' ]);
+        if (!stop && (symbol === undefined)) {
```

Comment:
`cancelOrder` in c# is not working as expected
```
 npm run cli.cs -- woofipro cancelOrder "1088411081" "WIF/USDC:USDC"          
Debugger attached.

> ccxt@4.3.14 cli.cs
> dotnet run --project "./cs/cli/cli.csproj" woofipro cancelOrder 1088411081 WIF/USDC:USDC

[
  1088411081,
  "WIF/USDC:USDC"
]
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.AggregateException: One or more errors occurred. (woofipro {"success":false,"code":-1000,"message":"The server encountered an internal error"})
 ---> ccxt.ExchangeError: woofipro {"success":false,"code":-1000,"message":"The server encountered an internal error"}
   at ccxt.woofipro.handleErrors(Object httpCode, Object reason, Object url, Object method, Object headers, Object body, Object response, Object requestHeaders, Object requestBody) in /Users/cjg/Git/ccxt10/ccxt/cs/ccxt/exchanges/woofipro.cs:line 2815
   at ccxt.Exchange.fetch(Object url2, Object method2, Object headers2, Object body2) in /Users/cjg/Git/ccxt10/ccxt/cs/ccxt/base/Exchange.cs:line 311
   at ccxt.Exchange.fetch2(Object path, Object api, Object method, Object parameters, Object headers, Object body, Object config) in /Users/cjg/Git/ccxt10/ccxt/cs/ccxt/base/Exchange.BaseMethods.cs:line 2868
   at ccxt.Exchange.callAsync(Object implicitEndpoint2, Object parameters) in /Users/cjg/Git/ccxt10/ccxt/cs/ccxt/base/Exchange.cs:line 364
   at ccxt.woofipro.v1PrivateDeleteOrder(Object parameters) in /Users/cjg/Git/ccxt10/ccxt/cs/ccxt/api/woofipro.cs:line 551
   at ccxt.woofipro.cancelOrder(Object id, Object symbol, Object parameters) in /Users/cjg/Git/ccxt10/ccxt/cs/ccxt/exchanges/woofipro.cs:line 1556
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean i
```

Reply:
@carlosmiei weird, I only got this error in cs. will take a look.

Reply:
The way we set content-type header is different in c#. Let me figure out how to set urlencode with empty body.